TicTacToe/test/tictactoe/TicTacToeTest.java

```diff
@@ -7,30 +7,168 @@
 
     @Test
     public void testIsWinner() {
+        
+        //empty
+        
+        TicTacToe testBoard = new TicTacToe(new char[][] {
+            {E, E, E},
+            {E, E, E},
+            {E, E, E}
+        });
+        
+        assertEquals(testBoard.isWinner(TicTacToe.X), false);
+        assertEquals(testBoard.isWinner(TicTacToe.O), false);
+        
+        //Vertical Line
+        
+        testBoard = new TicTacToe(new char[][] {
+            {X, O, E},
+            {E, O, E},
+            {E, O, X}
+        });
+        
+        assertEquals(testBoard.isWinner(TicTacToe.X), false);
+        assertEquals(testBoard.isWinner(TicTacToe.O), true);
+        
+        //Horizontal Line
+        
+        testBoard = new TicTacToe(new char[][] {
+            {E, O, E},
+            {X, X, X},
+            {E, O, O}
+        });
+        
+        assertEquals(testBoard.isWinner(TicTacToe.X), true);
+        assertEquals(testBoard.isWinner(TicTacToe.O), false);
+        
+        //diagonal line
+        
+        testBoard = new TicTacToe(new char[][] {
+            {E, E, X},
+            {O, X, X},
+            {X, O, O}
+        });
+        
+        assertEquals(testBoard.isWinner(TicTacToe.X), true);
+        assertEquals(testBoard.isWinner(TicTacToe.O), false);
+        
+        //tie
+        
+        testBoard = new TicTacToe(new char[][] {
+            {X, O, X},
+            {O, X, O},
+            {O, X, O}
+        });
+        
+        assertEquals(testBoard.isWinner(TicTacToe.X), false);
+        assertEquals(testBoard.isWinner(TicTacToe.O), false);
     }
 
     @Test
     public void testIsFull() {
+        
+        TicTacToe testBoard = new TicTacToe(new char[][] {
+            {X, O, X},
+            {O, O, X},
+            {X, X, E}
+        });
+        
+        assertEquals(testBoard.isFull(), false);
+        
+        testBoard = new TicTacToe(new char[][] {
+            {E, E, E},
+            {E, E, E},
+            {E, E, E}
+        });
+        
+        assertEquals(testBoard.isFull(), false);
+        
+        testBoard = new TicTacToe(new char[][] {
+            {X, X, X},
+            {O, O, X},
+            {X, X, O}
+        });
+        
+        assertEquals(testBoard.isFull(), true);
     }
 
     @Test
     public void testIsCat() {
+        
+        TicTacToe testBoard = new TicTacToe(new char[][] {
+            {X, O, X},
+            {O, O, X},
+            {X, X, E}
+        });
+        
+        assertEquals(testBoard.isCat(), false);
+        
+        testBoard = new TicTacToe(new char[][] {
+            {X, O, X},
+            {O, O, X},
+            {X, X, E}
+        });
+        
+        assertEquals(testBoard.isCat(), false);
```

Comment:
No tests for isCat == true.  No checks for when the board is full but there is a winner (which means isCat would be false).